src/main/java/bio/terra/service/tabulardata/google/bigquery/BigQuerySnapshotPdao.java

```diff
@@ -1416,11 +1416,22 @@ public interface Converter<T> {
     T convert(FieldValueList fieldValue);
   }
 
-  // WARNING: SQL string must be sanitized before calling this method
-  public <T> List<T> runQuery(String sql, Snapshot snapshot, Converter<T> converter) {
+  public <T> List<T> runQuery(
+      String query,
+      Map<String, String> userEnteredData,
+      Snapshot snapshot,
+      Converter<T> converter) {
+    QueryJobConfiguration queryConfig =
+        QueryJobConfiguration.newBuilder(query)
+            .setNamedParameters(
+                userEnteredData.entrySet().stream()
+                    .collect(
+                        Collectors.toMap(
+                            Map.Entry::getKey, e -> QueryParameterValue.string(e.getValue()))))
+            .build();
     try {
       final BigQueryProject bigQueryProject = BigQueryProject.from(snapshot);
-      final TableResult result = bigQueryProject.query(sql);
+      final TableResult result = bigQueryProject.getBigQuery().query(queryConfig);
```

Comment:
There is a version of `BigQueryProject.query()` which accepts `QueryParameterValue`s, so this can be written as:
```
    var values =
        userEnteredData.entrySet().stream()
            .collect(
                Collectors.toMap(
                    Map.Entry::getKey, e -> QueryParameterValue.string(e.getValue())));
    try {
      final BigQueryProject bigQueryProject = BigQueryProject.from(snapshot);
      final TableResult result = bigQueryProject.query(query, values);
```

Reply:
Since the map value is of type ? I need to cast to String. but that doesn't seem like the best option. Any other ideas? I could change the type of the map value back to String?

Reply:
Could we change the paramMap to be a Map<String, String> instead? Do we have a use case for a param that is not a string?

Reply:
As of now, no, its just for string

Reply:
OK, yes, you need `String` here, so the param map must be `<String, String>` after all. With that change, you no longer need the cast to `String`:
```
                    Map.Entry::getKey, e -> QueryParameterValue.string(e.getValue())));
```